src/http/transport-curl.cc

```diff
@@ -160,9 +160,9 @@ std::unique_ptr<Transport> TransportCurl::Clone() const {
 
 TransportCurl::~TransportCurl() {
   if (multi_handle_ && curl_) {
-    curl_multi_remove_handle(multi_handle_, curl_);
-    curl_multi_cleanup(multi_handle_);
-    curl_easy_cleanup(curl_);
+    curl_multi_remove_handle(multi_handle_, curl_);  // Remove easy handle
+    curl_easy_cleanup(curl_);           // Clean up all easy handles
+    curl_multi_cleanup(multi_handle_);  // Finally remove the multi handle
```

Comment:
Why reorder?

Reply:
I followed https://curl.se/libcurl/c/curl_multi_cleanup.html

Reply:
Which says:

> the order of cleaning up should be:
> 
> 1 - [curl_multi_remove_handle](https://curl.se/libcurl/c/curl_multi_remove_handle.html) before any easy handles are cleaned up
> 
> 2 - [curl_easy_cleanup](https://curl.se/libcurl/c/curl_easy_cleanup.html) can now be called independently since the easy handle is no longer connected to the multi handle
> 
> 3 - [curl_multi_cleanup](https://curl.se/libcurl/c/curl_multi_cleanup.html) should be called when all easy handles are removed